pkg/controllermanager/controller/project/project_activity_control.go

```diff
@@ -131,7 +131,7 @@ func (c *Controller) projectActivityObjectUpdate(ctx context.Context, oldObj, ne
 }
 
 func (c *Controller) getProjectKey(ctx context.Context, namespace string) string {
-	project, err := gutil.ProjectForNamespaceFromReader(ctx, c.gardenClient, namespace)
+	project, err := gutil.ProjectForNamespaceFromReader(ctx, c.cache, namespace)
```

Comment:
Just to double-check: Is it fine to use the cache here?

Reply:
Yes, definitely. We want to explicitly use the cache for reading objects in event handlers.